packages/backend/src/services/ValidationService/ValidationService.ts

```
@@ -208,198 +207,163 @@ export class ValidationService extends BaseService {
+                    fieldName,
+                }: {
+                    acc: CreateChartValidation[];
+                    fieldIds: string[];
+                    fieldId: string;
+                } & Pick<
+                    CreateChartValidation,
+                    'error' | 'errorType' | 'fieldName'
+                >) => {
+                    if (!fieldIds?.includes(fieldId)) {
+                        return [
+                            ...acc,
+                            {
+                                ...commonValidation,
+                                errorType,
+                                error,
+                                fieldName,
+                            },
+                        ];
+                    }
+                    return acc;
+                };
 
+                const dimensionErrors = dimensions.reduce<
+                    CreateChartValidation[]
+                >(
+                    (acc, field) =>
+                        containsFieldId({
+                            acc,
+                            fieldIds: [
+                                ...availableDimensionIds,
+                                ...availableCustomDimensionIds,
+                            ],
+                            fieldId: field,
+                            error: `Dimension error: the field '${field}' no longer exists`,
+                            errorType: ValidationErrorType.Dimension,
+                            fieldName: field,
+                        }),
+                    [],
+                );
+                const metricErrors = metrics.reduce<CreateChartValidation[]>(
+                    (acc, field) =>
+                        containsFieldId({
+                            acc,
+                            fieldIds: [...availableMetricIds, ...customMetrics],
+                            fieldId: field,
+                            error: `Metric error: the field '${field}' no longer exists`,
```

should we return a different error message if it is a custom metric ?